src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -235,6 +247,7 @@ private void LoadItems(PackageItemListViewModel selectedPackageItem, Cancellatio
                 UpdateCheckBoxStatus();
 
                 LoadItemsCompleted?.Invoke(this, EventArgs.Empty);
+                Items.Remove(_loadingStatusIndicator);
```

Comment:
Wrap this in a finally {} block from the outermost try/catch ?
We call SetError() on the _loadingStatusBar, but does that hide the Indicator or does it stay displayed infinitely?

Reply:
If SetError already hides the indicator, then this line should be moved inside the outermost try {} logic so that it only removes the item if no Exception occurred.

Reply:
Nice catch!

> Wrap this in a finally {} block from the outermost try/catch ?

The idea is to add the indicator at the very beginning and remove it at the very end of the method execution. I prefer to leave it outside of the try cath

> We call SetError() on the _loadingStatusBar, but does that hide the Indicator or does it stay displayed infinitely?

If we look futher, `_loadingStatusBar.SetError()` only sets an internal error message. I didn't find any bindings or references of the error message outside of the `LoadingStatusIndicator` class. So, I will remove that property.

> If SetError already hides the indicator, then this line should be moved inside the outermost try {} logic so that it only removes the item if no Exception occurred.

For the reasons stated above, I prefer to leave it outside.

This lead me to discover an issue with this PR. See comments added.

Reply:
Nice catch!

> Wrap this in a finally {} block from the outermost try/catch ?

The idea is to add the indicator at the very beginning and remove it at the very end of the method execution. I prefer to leave it outside of the try cath

> We call SetError() on the _loadingStatusBar, but does that hide the Indicator or does it stay displayed infinitely?

If we look futher, `_loadingStatusBar.SetError()` only sets an internal error message. I didn't find any bindings or references of the error message outside of the `LoadingStatusIndicator` class. So, I will remove that property.

> If SetError already hides the indicator, then this line should be moved inside the outermost try {} logic so that it only removes the item if no Exception occurred.

For the reasons stated above, I prefer to leave it outside.

This lead me to discover an issue with this PR. See comments added.

Reply:
I managed to solve the issue documented in the previous TODO. At the end, I moved the loading indicator removal to a `finally` block